### PR TITLE
fix: query tx events with `>=` and `<=` operators

### PR DESCRIPTION
**Pull request by rootulp**
Part of https://github.com/celestiaorg/celestia-app/issues/2098

### Before
```
$ curl http://localhost:1317/cosmos/tx/v1beta1/txs?events=tx.height%3C=10
{
"code": 3,
"message": "invalid event; event tx.height\u003c=10 should be of the format: {eventType}.{eventAttribute}={value}",
"details": [
]
}%
```

### After

```shell
$ curl http://localhost:1317/cosmos/tx/v1beta1/txs?events=tx.height%3C=10
{
  "txs": [
  ],
  "tx_responses": [
  ],
  "pagination": null,
  "total": "0"
}%
```

## FLUPs

1. Cut a release of this repo and bump in celestia-app
1. Get a fix for this issue merged to upstream cosmos-sdk https://github.com/cosmos/cosmos-sdk/pull/16994 and pull it here

### CODE REVIEW

File: x/auth/tx/service_test.go
Comment (SweeXordious) [inline x/auth/tx/service_test.go:187]: [Non blocking][Question]
Does it make sense to add tests also for the other operators supported by core?

https://github.com/celestiaorg/celestia-core/blob/3e636bb02f48cc02b11f4aea3dd1494ef8e62c91/libs/pubsub/query/query.go#L65-L84

Even if they're caught by the regex, so that upcoming changes don't end up breaking the API again

**Review by staheri14 — APPROVED** [review]

**Review by SweeXordious — APPROVED** [review]

File: x/auth/tx/service_test.go
Comment (rootulp) [reply x/auth/tx/service_test.go:187]: Wow great question! I think we should do this in a follow-up b/c I don't want to dismiss approvals. TBH I'd be a little surprised if "EXISTS" and "CONTAINS" work b/c the current regex doesn't special case for those.

**Pull request merged**
